rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestIntegration.java

```diff
@@ -484,32 +539,34 @@ void callUpdateCollectionsThrowsCouldNotAdjustDownwards() throws IOException {
 
         org.ethereum.core.Block rskCurrentBlock = blocks.get(9);
         Transaction tx = Transaction
-                .builder()
-                .nonce(NONCE)
-                .gasPrice(GAS_PRICE)
-                .gasLimit(GAS_LIMIT)
-                .destination(Hex.decode(TO_ADDRESS))
-                .data(Hex.decode(DATA))
-                .chainId(Constants.REGTEST_CHAIN_ID)
-                .value(DUST_AMOUNT)
-                .build();
+            .builder()
+            .nonce(NONCE)
+            .gasPrice(GAS_PRICE)
+            .gasLimit(GAS_LIMIT)
+            .destination(Hex.decode(TO_ADDRESS))
+            .data(Hex.decode(DATA))
+            .chainId(Constants.REGTEST_CHAIN_ID)
+            .value(DUST_AMOUNT)
+            .build();
         tx.sign(new ECKey().getPrivKeyBytes());
 
         BridgeStorageProvider providerForSupport = new BridgeStorageProvider(
-                track,
-                PrecompiledContracts.BRIDGE_ADDR,
-                bridgeConstants.getBtcParams(),
-                activationsBeforeForks
-        );
-        BridgeSupport bridgeSupport = getBridgeSupport(
-                bridgeConstants,
-                providerForSupport,
-                track,
-                mock(BridgeEventLogger.class),
-                rskCurrentBlock,
-                null
+            track,
+            PrecompiledContracts.BRIDGE_ADDR,
+            bridgeConstants.getBtcParams(),
+            activationsBeforeForks
         );
 
+        federationStorageProvider = createFederationStorageProvider(track);
+        FederationSupport federationSupport = federationSupportBuilder
+            .withFederationConstants(federationConstants)
+            .withFederationStorageProvider(federationStorageProvider)
+            .withActivations(activationsBeforeForks)
+            .withRskExecutionBlock(rskCurrentBlock)
+            .build();
```

Comment:
Since you are using a bunch of times this builder. Maybe have a private method that returns this?

Reply:
yes, that's a good idea.
Since we are using bridge support and federation support builders in many test files i thought about creating a "builder helper" class to avoid having the same private method in all of them.
Lets discuss it